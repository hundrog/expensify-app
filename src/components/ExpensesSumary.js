import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import getExpensesTotal from "../selectors/expenses-total";
import selectExpenses from "../selectors/expenses";

export const ExpensesSumary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? "expense" : "expenses";
  const formatedTotall = numeral(expensesTotal / 100).format("$0,0.00");
  return (
    <div>
      <h1>
        {`Viewing ${expenseCount} ${expenseWord} totalling ${formatedTotall}`}
      </h1>
    </div>
  );
};

const mapStateToProps = state => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: getExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSumary);
