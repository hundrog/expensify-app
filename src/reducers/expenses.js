// Expenses Reducer
const expensesReducerDefaultState = [];

const expensesReducer = (state = expensesReducerDefaultState, action) => {
  switch (action.type) {
    case "ADD_EXPENSE":
      return [...state, action.expense]; //array spreading, take an array and add at the end or begining
    case "REMOVE_EXPENSE":
      // So... we take the expense, destructure the id, then compare the id to filter it
      return state.filter(({ id }) => id !== action.id);
    case "EDIT_EXPENSE":
      return state.map(expense => {
        if (expense.id === action.id) {
          // object spreading, take the object the override
          return {
            ...expense,
            ...action.updates
          };
        } else {
          return expense;
        }
      });
    default:
      return state;
  }
};

export default expensesReducer;
