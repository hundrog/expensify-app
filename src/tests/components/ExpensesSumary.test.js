import React from "react";
import { shallow } from "enzyme";
import expenses from "../fixtures/expenses";
import { ExpensesSumary } from "../../components/ExpensesSumary";

test("should render correctly with one expense", () => {
  const wrapper = shallow(
    <ExpensesSumary expenseCount={1} expensesTotal={2010} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should render correctly with multiple expenses", () => {
  const wrapper = shallow(
    <ExpensesSumary expenseCount={4} expensesTotal={19902000} />
  );
  expect(wrapper).toMatchSnapshot();
});
