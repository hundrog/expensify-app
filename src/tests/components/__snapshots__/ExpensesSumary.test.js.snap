// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 4 expenses totalling $199,020.00
  </h1>
</div>
`;

exports[`should render correctly with one expense 1`] = `
<div>
  <h1>
    Viewing 1 expense totalling $20.10
  </h1>
</div>
`;
